test/libsolidity/SemanticTest.cpp

```diff
@@ -105,16 +107,25 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 {
 	TestResult result = TestResult::Success;
 	bool compileViaYul = m_runWithYul || m_enforceViaYul;
+	bool compileToEwasm = m_runWithEwasm || m_enforceCompileToEwasm;
 
 	if (m_runWithoutYul)
 		result = runTest(_stream, _linePrefix, _formatted, false, false);
 
 	if (compileViaYul && result == TestResult::Success)
 		result = runTest(_stream, _linePrefix, _formatted, true, false);
 
-	if (m_runWithEwasm && result == TestResult::Success)
-		result = runTest(_stream, _linePrefix, _formatted, true, true);
-
+	if (compileToEwasm && result == TestResult::Success)
+	{
+		try
+		{
+			result = runTest(_stream, _linePrefix, _formatted, true, true);
+		}
+		catch (...) {
+			if (!m_enforceCompileToEwasm || m_runWithEwasm)
```

Comment:
Isn't this redundant? `compileToEwasm` is defined as one of those, so if `!m_enforce` is false, `m_runWithEwasm` must be true

Reply:
ah yes.. thats strange